pkg/auth/zcapld/service.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package zcapld
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	cryptoapi "github.com/hyperledger/aries-framework-go/pkg/crypto"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/util/signature"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/vdr/fingerprint"
+	"github.com/igor-pavlenko/httpsignatures-go"
+	"github.com/trustbloc/edge-core/pkg/zcapld"
+)
+
+// Service to provide zcapld functionality.
+type Service struct {
+	keyManager kms.KeyManager
+	crypto     cryptoapi.Crypto
+}
+
+// New return zcap service.
+func New(keyManager kms.KeyManager, crypto cryptoapi.Crypto) *Service {
+	return &Service{keyManager: keyManager, crypto: crypto}
+}
+
+// CreateDIDKey create did key.
+func (s *Service) CreateDIDKey() (string, error) {
+	signer, err := signature.NewCryptoSigner(s.crypto, s.keyManager, kms.ED25519)
+	if err != nil {
+		return "", fmt.Errorf("failed to create crypto signer: %w", err)
+	}
+
+	_, didKeyURL := fingerprint.CreateDIDKey(signer.PublicKeyBytes())
+
+	return didKeyURL, nil
+}
+
+// SignHeader sign header.
+func (s *Service) SignHeader(req *http.Request, capabilityBytes []byte) (*http.Header, error) {
+	capability, err := zcapld.ParseCapability(capabilityBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	compressedZcap, err := compressZCAP(capability)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set(zcapld.CapabilityInvocationHTTPHeader,
+		fmt.Sprintf(`zcap capability="%s",action="%s"`, compressedZcap, "read"))
```

Comment:
Doesn't hub-kms (in ops key mode) also need to "write" to EDV vaults?

Reply:
done

Reply:
I think it needs to read or write, depending on the action it wants to do at a particular point in time. That's why I asked if it _also_ needs to write to EDV vaults.

Reply:
@llorllale yes it's fixed in this PR https://github.com/trustbloc/edv/pull/155

Reply:
Yes. Now, hub-kms needs to set either "read" or "write" depending on the operation (eg. writing a new doc when creating a new key vs reading an existing document when something needs to be signed). It can't be hard-coded.

Reply:
done